shell.nix

```diff
@@ -1,3 +1,39 @@
-(import (fetchTarball https://github.com/edolstra/flake-compat/archive/master.tar.gz) {
-  src = builtins.fetchGit ./.;
-}).shellNix
+{ pkgs ? import <nixpkgs> {} }:
+pkgs.mkShell {
+    buildInputs = [ pkgs.go   pkgs.postgresql   pkgs.mailhog];
+
+    src = pkgs.fetchFromGitHub {
+       owner = "privacybydesign";
+       repo = "irmago";
+       rev = "0c153a97d013162688e77f104c65bd246b5d5323";
+       sha256 = "sha256-Y+DUFOlgmyHloQwq0MTs/oabLf+8NHVElguoM9EVvmo=";
+     };
+
+     shellHook = ''
+       export PGHOST=./postgres
+       export PGDATA=$PGHOST/data
+       export PGDATABASE=postgres
+       export PGLOG=$PGHOST/postgres.log
```

Comment:
Setting the environment variables in the shellHook is good.
Is the PGLOG var only needed for postgres ?
If it is you don't need to set another environment variable and can directly replace the value inside the pg_ctl call

However all the mutable changes below should be part of the readme, not part of the hook.
The way you have it setup here, those changes run on every time you cd in the directory (if direnv is installed for example). There could be ways to mitigate this, but setups steps should most likely be part of the readme.

Reply:
You could make the whole setup procedure a bash function and then the user can just execute `init-postgres`, or something similar. Then it is not executed on every session.

Reply:
The initialisation of postgreSql and MailHog is only for testing?. If we remove the test componet in the flake then we can seperate it. I was of the understanding that there is a requirement for testing in an fully automated manner..

Reply:
Postgres is needed for dev environment as well right ?
a bash function sounds good to me.